tests/python_api/audio_unit_test.py

```diff
@@ -0,0 +1,178 @@
+        #comparing the absolute difference between the output value (out_val) and the reference value (ref_val) with a tolerance threshold of 1e-20.
+        if not invalid_comparison and np.abs(out_val - ref_val) < 1e-20:
+            matched_indices += 1
+
+    # Print results
+    print(f"Results for {case_name}:")
+    if matched_indices == roi_data[0] and matched_indices != 0:
+        print("PASSED!")
+        test_results[case_name] = "PASSED"
+    else:
+        print("FAILED!")
+        test_results[case_name] = "FAILED"
+
+@pipeline_def(seed=seed)
+def audio_decoder_pipeline(path):
+    audio, labels = fn.readers.file(file_root=path)
+    return fn.decoders.audio(
+        audio,
+        file_root=path,
+        downmix=False,
+        shard_id=0,
+        num_shards=1,
+        stick_to_shard=False)
+
+def main():
+    args = parse_args()
+
+    audio_path = args.audio_path
+    rocal_cpu = False if args.rocal_gpu else True
+    batch_size = args.batch_size
+    test_case = args.test_case
+    qa_mode = args.qa_mode
+    num_threads = 1
+    device_id = 0
+    rocal_data_path = os.environ.get("ROCAL_DATA_PATH")
+
+    case_list = list(test_case_augmentation_map.keys())
+
+    if test_case is not None: 
+        if test_case not in case_list:
+            print(" Invalid Test Case! ")
+            exit()
+        else:
+            case_list = [test_case]
+
+    if args.display:
+        try:
+            path = "OUTPUT_FOLDER/AUDIO_READER"
+            isExist = os.path.exists(path)
+            if not isExist:
+                os.makedirs(path)
+        except OSError as error:
+            print(error)
+
+    if rocal_data_path is None:
+        print("Need to export ROCAL_DATA_PATH")
+        sys.exit()
+    if not rocal_cpu:
+        print("The GPU support for Audio is not given yet. running on cpu")
+        rocal_cpu = True
+    if audio_path == "":
+        audio_path = f'{rocal_data_path}/rocal_data/audio/wav/'
+    else:
+        print("QA mode is disabled for custom audio data")
+        qa_mode = 0
+    if qa_mode and batch_size != 1:
+        print("QA mode is enabled. Batch size is set to 1.")
+        batch_size = 1
+
+    print("*********************************************************************")
+    test_results = {}
+    for case in case_list:
+        case_name = test_case_augmentation_map.get(case)
+        if case_name == "audio_decoder":
+            audio_pipeline = audio_decoder_pipeline(batch_size=batch_size, num_threads=num_threads, device_id=device_id, rocal_cpu=rocal_cpu, path=audio_path)
+        audio_pipeline.build()
+        audioIteratorPipeline = ROCALAudioIterator(audio_pipeline, auto_reset=True)
+        cnt = 0
+        import timeit
+        start = timeit.default_timer()
+        # Enumerate over the Dataloader
+        for e in range(int(args.num_epochs)):
+            print("Epoch :: ", e)
+            torch.set_printoptions(threshold=5000, profile="full", edgeitems=100)
```

Comment:
Are we using threshold and edgeitems? What's the use?

Reply:
used for displying the tensor completely

Reply:
Generally Pytorch prints only a summary replresentation of the tensor.